src/Setup/WixBA/RootView.xaml

```diff
@@ -1,125 +1,487 @@
+                                        AutomationProperties.HelpText="Click to try again"
+                                        AutomationProperties.Name="Try Again?"
+                                        Command="{Binding TryAgainCommand}"
+                                        Content="Try Again?"
+                                        Style="{DynamicResource ActionButtonStyle}"
+                                        Visibility="{Binding TryAgainEnabled,
+                                                             Converter={StaticResource BooleanToVisibilityConverter}}" />
+                            </StackPanel>
+
+                            <!--  Final Actions - only one of these is expected to be displayed, if any  -->
+                            <StackPanel x:Name="CloseBtnDcIsland"
+                                        HorizontalAlignment="Right"
+                                        DataContext="{Binding InstallationViewModel}"
+                                        Style="{DynamicResource ActionBtnStkPnlStyle}">
+                                <Button x:Name="CloseButton"
+                                        AutomationProperties.HelpText="Click to Close"
+                                        AutomationProperties.Name="Close"
+                                        Command="{Binding CloseCommand}"
+                                        Content="Close"
+                                        Style="{DynamicResource FinalActionButtonStyle}"
+                                        Visibility="{Binding CloseEnabled,
+                                                             Converter={StaticResource BooleanToVisibilityConverter}}" />
+                            </StackPanel>
+                            <StackPanel x:Name="CancelBtnDcIsland"
+                                        HorizontalAlignment="Right"
+                                        DataContext="{Binding}"
+                                        Style="{DynamicResource ActionBtnStkPnlStyle}">
+                                <Button AutomationProperties.HelpText="Press to Cancel"
+                                        AutomationProperties.Name="Cancel"
+                                        Command="{Binding CancelCommand}"
+                                        Content="Cancel"
+                                        Style="{DynamicResource FinalActionButtonStyle}"
+                                        Visibility="{Binding CancelEnabled,
+                                                             Converter={StaticResource BooleanToVisibilityConverter}}" />
+                            </StackPanel>
+
+                        </StackPanel>
+                    </Grid>
+                </DockPanel>
+
+                <StackPanel x:Name="FinalStatusArea"
+                            HorizontalAlignment="Stretch"
+                            DockPanel.Dock="Bottom">
```

Comment:
The main thing I don't like here is that in order for this StackPanel to be above the previous StackPanel in the UI, it has to be below it in the XAML.